connectors/github-pr/src/test/java/com/vmware/connectors/github/pr/GithubPrControllerTests.java

```diff
@@ -0,0 +1,710 @@
+        requestCards(GITHUB_AUTH_TOKEN, fromFile("requests/valid/cards/empty-pr-urls.json"))
+                .andExpect(status().isOk())
+                .andExpect(content().contentTypeCompatibleWith(APPLICATION_JSON))
+                .andExpect(
+                        content().string(
+                                JsonReplacementsBuilder
+                                        .from(fromFile("responses/success/cards/empty-pr-urls.json"))
+                                        .buildForCards()
+                        )
+                );
+    }
+
+    @Test
+    public void testRequestCardsMissingPrUrlsSuccess() throws Exception {
+        requestCards(GITHUB_AUTH_TOKEN, fromFile("requests/valid/cards/missing-pr-urls.json"))
+                .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    public void testRequestCardsEmptyTokens() throws Exception {
+        requestCards(GITHUB_AUTH_TOKEN, fromFile("requests/invalid/cards/empty-tokens.json"))
+                .andExpect(status().isBadRequest())
+                .andExpect(content().contentTypeCompatibleWith(APPLICATION_JSON))
+                .andExpect(content().json(fromFile("responses/error/cards/empty-tokens.json"), false));
+    }
+
+    @Test
+    public void testRequestCardsMissingTokens() throws Exception {
+        requestCards(GITHUB_AUTH_TOKEN, fromFile("requests/invalid/cards/missing-tokens.json"))
+                .andExpect(status().isBadRequest())
+                .andExpect(content().contentTypeCompatibleWith(APPLICATION_JSON))
+                .andExpect(content().json(fromFile("responses/error/cards/missing-tokens.json"), false));
+    }
+
+    /////////////////////////////
+    // Approve Action
+    /////////////////////////////
+
+    @Test
+    public void testApproveActionUnauthorized() throws Exception {
+        mockGithub.expect(requestTo(any(String.class)))
+                .andRespond(withUnauthorizedRequest());
+
+        approve(GITHUB_AUTH_TOKEN)
+                .andExpect(status().isBadRequest())
+                .andExpect(header().string("X-Backend-Status", "401"));
+    }
+
+    @Test
+    public void testApproveAuthHeaderMissing() throws Exception {
+        approve(null)
+                .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    public void testApproveActionSuccess() throws Exception {
+        String fakeResponse = fromFile("fake/actions/approve/success.json");
+
+        String expected = fromFile("responses/actions/approve/success.json");
+
+        mockGithub.expect(requestTo("https://api.github.com/repos/vmware/test-repo/pulls/99/reviews"))
+                .andExpect(header(AUTHORIZATION, "Bearer " + GITHUB_AUTH_TOKEN))
+                .andExpect(method(POST))
+                .andExpect(MockRestRequestMatchers.content().contentType(APPLICATION_JSON))
+                // Note: This doesn't verify that we don't get an explicit null, see SPR-16339
```

Comment:
This was really unfortunate, so I created [SPR-16339](https://jira.spring.io/browse/SPR-16339) for it.